### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by gurellia53**

### CODE REVIEW

File: 2024 Code/src/main/java/frc/robot/util/CalculateAngle.java [hunk 1]
```diff
@@ -196,6 +179,15 @@ public double InterpolateAngle() {
         p3 = new Point2D.Double(x1, y2);
         p4 = new Point2D.Double(x2, y2);
 
+        System.err.println(p3.x + "," + p3.y + "______________________________________" + p4.x + "," + p4.y);
```
Comment (TrentAgnos) [inline 2024 Code/src/main/java/frc/robot/util/CalculateAngle.java:182]: This should definitely be disabled before merging

File: 2024 Code/src/main/java/frc/robot/commands/AutoAlignCommand.java [hunk 2]
```diff
@@ -35,7 +35,8 @@ public void execute() {
         if (SubsystemContainer.limelightSubsystem.getTv()) {
             currentYaw = SubsystemContainer.swerveSubsystem.getYaw();
             offsetYaw = SubsystemContainer.limelightSubsystem.getTx();
-            SubsystemContainer.swerveSubsystem.setRobotYaw(currentYaw - offsetYaw);
+            double[] pose = SubsystemContainer.limelightSubsystem.getBotPose();
+            SubsystemContainer.swerveSubsystem.setRobotYaw(currentYaw - offsetYaw + (pose[5] * 0.05));
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/commands/AutoAlignCommand.java:39]: can you name pose[5]? better yet, call a properly named getter in the limelight subsystem to get just this double

Comment (gurellia53) [reply 2024 Code/src/main/java/frc/robot/commands/AutoAlignCommand.java:39]: maybe put this math in a method with a good name so we understand what this is

File: 2024 Code/src/main/java/frc/robot/commands/MagicAmpChargeCommand.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ public void initialize() {
   @Override
   public void execute() {
     m_armSubsystem.setArmReferenceAngle(Constants.ArmConstants.AMP_POSE);
-    m_shooterSubsystem.setTargetRPM(1000);
+    m_shooterSubsystem.setTargetRPM(1500);
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/commands/MagicAmpChargeCommand.java:39]: this needs to be a constant

File: 2024 Code/src/main/java/frc/robot/subsystems/IntakeSubsystem.java [hunk 4]
```diff
@@ -205,5 +205,7 @@ public void periodic() {
         m_indexBeam = !m_indexBeamBreak.get();
         m_intakeBeam = !m_intakeBeamBreak.get();
 
+        System.out.println(m_intakeBeam);
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/subsystems/IntakeSubsystem.java:208]: nope

File: 2024 Code/src/main/java/frc/robot/commands/ShootManualCommand.java [hunk 5]
```diff
@@ -32,11 +35,6 @@ public void initialize() {
     public void execute() {
         m_armSubsystem.setArmReferenceAngle(m_angle);
         m_shooterSubsystem.setTargetRPM(m_rpm);
-        if (m_armSubsystem.getInPosition() && m_shooterSubsystem.approveShoot()) {
-            m_intakeSubsystem.runIndexShoot();
-        } else {
-            m_intakeSubsystem.stopIndex();
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/commands/ShootManualCommand.java:38]: should stopIndex still be here?

File: 2024 Code/src/main/java/frc/robot/commands/ShootManualCommand.java [hunk 6]
```diff
@@ -35,6 +35,7 @@ public void initialize() {
     public void execute() {
         m_armSubsystem.setArmReferenceAngle(m_angle);
         m_shooterSubsystem.setTargetRPM(m_rpm);
+        m_intakeSubsystem.runIndexFeed();
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/commands/ShootManualCommand.java:38]: i might be misunderstanding, but shouldn't this command stop the index motors?

Comment (gurellia53) [reply 2024 Code/src/main/java/frc/robot/commands/ShootManualCommand.java:38]: oh this method does that. nvm. nice

File: 2024 Code/src/main/java/frc/robot/commands/ShooterInterpolateCommand.java [hunk 7]
```diff
@@ -35,12 +35,8 @@ public void initialize() {
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
-    m_shooterSubsystem.setTargetRPM(m_RPMCalculate.InterpolateRPM());
-    if (m_armSubsystem.getInPosition() && m_shooterSubsystem.approveShoot()) {
-      m_intakeSubsystem.runIndexShoot();
-    } else {
-      m_intakeSubsystem.stopIndex();
-    }
+    m_shooterSubsystem.setTargetRPM(2000);
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/commands/ShooterInterpolateCommand.java:38]: 2000?

File: 2024 Code/src/main/java/frc/robot/commands/AutoAlignSequentialCommand.java [hunk 8]
```diff
@@ -25,13 +26,32 @@ public void initialize() {
...
+        if (!SubsystemContainer.swerveSubsystem.isRedAlliance) {
+            if (pose[5] > 0) {
+                pose[5] -= 180;
+            } else {
+                pose[5] += 180;
+            }
+
+        }
+        System.out.println("tx " + offsetYaw);
+        SubsystemContainer.swerveSubsystem.autonRotateSwerve(currentYaw - offsetYaw + (pose[5] * 0.04));
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/commands/AutoAlignSequentialCommand.java:38]: currentYaw - offsetYaw + (pose[5] * 0.04) is common code and should be moved to the subsystem

**Review by ViennaRossmanith — APPROVED** [review]
epic

**Pull request merged**